src/pages/pretest-checks/pretest-checks.ts

```diff
@@ -1,25 +1,53 @@
+import { EyesightFaultRecordingModalPage } from './../eyesight-fault-recording-modal/eyesight-fault-recording-modal';
 import { Component } from '@angular/core';
-import { NavController, NavParams } from 'ionic-angular';
+import { NavController, ModalController } from 'ionic-angular';
 import { PolicyDataPage } from '../policy-data/policy-data';
 import { InitiateDrivingPeriodPage } from '../initiate-driving-period/initiate-driving-period';
 import { EndTestReasonPage } from '../end-test-reason/end-test-reason';
 import { Page } from 'ionic-angular/navigation/nav-util';
+import { TellMeModalComponent } from '../../components/tell-me-modal/tell-me-modal';
 
 @Component({
   selector: 'page-pretest-checks',
-  templateUrl: 'pretest-checks.html',
+  templateUrl: 'pretest-checks.html'
 })
 export class PretestChecksPage {
-
   endTestReasonPage: Page = EndTestReasonPage;
   initiateDrivingPeriodPage: Page = InitiateDrivingPeriodPage;
   policyDataPage: Page = PolicyDataPage;
+  tellMeQuestionId: string;
+  eyeSightSeriousFault = false;
+  disableBackdropDismissModalOption = { enableBackdropDismiss: false };
 
-  constructor(public navCtrl: NavController, public navParams: NavParams) {
-  }
+  constructor(public navCtrl: NavController, private modalCtrl: ModalController) {}
+
+  showTellMeOptions() {
+    const tellMeQuestionModal = this.modalCtrl.create(
+      TellMeModalComponent,
+      { selectedTellMeQuestionId: this.tellMeQuestionId },
+      this.disableBackdropDismissModalOption
+    );
+
+    tellMeQuestionModal.onDidDismiss((selectedTellMeQuestionId, role: string) => {
+      if (role !== 'dismiss') {
+        this.tellMeQuestionId = selectedTellMeQuestionId;
+      }
+    });
 
-  ionViewDidLoad() {
-    console.log('ionViewDidLoad PretestChecksPage');
+    tellMeQuestionModal.present();
   }
 
+  showEyesightFaultRecordingModal() {
```

Comment:
I think this will eventually move to the abandon test screen (and pop up based on why you press abandon), we can leave it here for now though.

Reply:
the DE might first record the eyesight fault and then eg car registration number. To end test there is a special button at the button, but its good that you are thinking about such things !